Assets/MirageXR/ContentTypes/VirtualInstructor/Editors/Mobile/AudioStreamPlayer.cs

```diff
@@ -0,0 +1,216 @@
+    /// </summary>
+    private TMP_Text _currentTimeText;
+    private TMP_Text _durationText;
+
+    /// <summary>
+    /// Represents a play button used for audio streaming.
+    /// </summary>
+    private Button _playButton;
+    /// <summary>
+    /// Reference to the forward button in the UI.
+    /// </summary>
+    private Button _forward;
+
+    /// <summary>
+    /// Reference to the backward button in the UI.
+    /// </summary>
+    private Button _backward;
+
+    /// <summary>
+    /// Represents the game object for the play button in the UI.
+    /// </summary>
+    public GameObject play;
+    public GameObject pause;
+
+
+    /// <summary>
+    /// Represents an AI model for the AiServices.
+    /// </summary>
+    private AIModel _model;
+    private TMP_Text[] _textComponents;
+
+    /// <summary>
+    /// An array of Button objects representing the interactive buttons in the AudioStreamPlayer script. 
+    /// </summary>
+    private Button[] _buttons;
+
+    /// <summary>
+    /// Represents the current playing state of the AudioStreamPlayer.
+    /// </summary>
+    private bool _isPlaying;
+
+
+    /// <summary>
+    /// Initializes the AudioStreamPlayer by assigning references to its required components.
+    /// </summary>
+    void Awake()
+    {
+        
+        _audioSource = GetComponent<AudioSource>();
+        _progressSlider = gameObject.GetComponentInChildren<Slider>(); // ist null !
+        _textComponents = gameObject.GetComponentsInChildren<TMP_Text>();
+        _currentTimeText = _textComponents[1];
+        _durationText = _textComponents[2];
+        _buttons = gameObject.GetComponentsInChildren<Button>();
+        _backward = _buttons[1];
+        _playButton = _buttons[2];
+        _forward = _buttons[3];
+    }
+
+
+    /// <summary>
+    /// Sets up the AudioStreamPlayer component with the specified AIModel.
+    /// </summary>
+    /// <param name="model">The AIModel to use for setup.</param>
+    public async void Setup(AIModel model)
+    {
+        _model = model;
+        if (_textComponents.Length == 3)
+        {
+            _textComponents[0].text = _model.Name;
+            _currentTimeText.text = "0.00";
+            _audioSource.clip = await LoadAudioAsync();
+            play.gameObject.SetActive(true);
+            _playButton.onClick.AddListener(PlayAudio);
+            _backward.onClick.AddListener(MoveBackward);
+            _forward.onClick.AddListener(MoveForward);
+            _progressSlider.onValueChanged.AddListener(OnSliderValueChanged);
+            _durationText.text = _audioSource.clip.length.ToString("F2", CultureInfo.CurrentCulture);
+        }
+    }
+
+    /// <summary>
+    /// Moves the audio playback forward by 10% of the total duration.
+    /// </summary>
+    private void MoveForward()
+    {
+        if (_audioSource != null)
+        {
+            float newTime = _audioSource.time + (_audioSource.clip.length * 0.1f);
```

Comment:
pls move all the numbers to "const float..."

Reply:
I don't get that one. To my understanding, a const has to be declared during compilation and can't be changed. I use it here to update the slider and timer text. So I have to update in the update method and that trows compiler errors.

Reply:
I mean numbers that do not change during the execution of the program. Like 0.1f (10%) and so on.

Reply:
Ah okay! Give me 15 minutes.

Reply:
Done.